sandy-city-utah/active-directory-api
# fix(general): extract shared features

### What Changed?
  - extract logging, csv export, and shared types to separate packages (removed from this project)
  - add general request logger to make it easier to see what the request came through as
  - redirect some imports to be coming from packages
  - add a not nullable constraint to the department column for users
  - **Breaking changes are mentioned below**

### Issues Addressed:
  - P21312-15

### Issue Testing:
  - Manual Developer Testing

### Test Results:
1. Run the application
1. Make requests to the api
1. See that it now logs the request information in the logs (can use an option to see it in the console as well)
1. See that the applications using the api were not affected as no api usability changes were made

### Additional Information:
**Breaking change:**
Until we resolve how the code will be shared, each of those extracted packages will either have to be pulled from the github repo or cloned locally and then linked.  Instructions on how to do that are found in those packages' readmes and not added to this readme (yet) for the expectation of speedily resolving the package setup.

Currently suggesting to use the GitHub package registry as it's free for our limited uses (for now).
